### PR TITLE
feat: migrates keywords and ECs to saved search

### PR DESCRIPTION
**Pull request by as1729**
## Description
- Adds a script that migrates keywords into saved searches.
- Can be run as `yarn run migrate_saved_searches` with the optional `--dry-run` flag

#### Dry Run - Note: Logs are truncated as most agencies locally do not have users
```
bash-5.1# yarn run migrate_saved_searches --dry-run
yarn run v1.22.17
$ node src/db/saved_search_migration.js --dry-run
DRY RUN :: Begin migrating legacy agency criteria to saved searches
DRY RUN :: Migrating agency criteria for agency 0
DRY RUN :: No agency criteria to migrate for agency 400
...
DRY RUN :: Migrating agency criteria for agency 380
DRY RUN :: Migrating agency criteria for agency 381
DRY RUN :: Migrating agency criteria for agency 382
DRY RUN :: Migrating agency criteria for agency 383
DRY RUN :: Migrating agency criteria for users 12 belonging to agency 386
DRY RUN :: Migrating agency criteria for users 9,10,11 belonging to agency 113
DRY RUN :: Migrating agency criteria for users 1,2,3,4,5,17,18 belonging to agency 0
DRY RUN :: No agency criteria to migrate for users 14,15 belonging to agency 400
DRY RUN :: Migrating agency criteria for users 6,7,8,13 belonging to agency 384
DRY RUN :: No users to migrate for agency 1
...
DRY RUN :: No users to migrate for agency 383
DRY RUN :: No users to migrate for agency 385
DRY RUN :: No users to migrate for agency 401
DRY RUN :: No users to migrate for agency 402
DRY RUN :: Would have inserted approximately 15 saved searches. Note: there may be duplicates.
DRY RUN :: Done migrating legacy agency criteria to saved searches
Done in 2.39s.
bash-5.1# 
```
#### Live Run - Note: Logs are truncated as most agencies locally do not have users
```
bash-5.1# yarn run migrate_saved_searches
yarn run v1.22.17
$ node src/db/saved_search_migration.js
Begin migrating legacy agency criteria to saved searches
Migrating agency criteria for agency 0
No agency criteria to migrate for agency 400
Migrating agency criteria for agency 401
Migrating agency criteria for agency 402
Migrating agency criteria for agency 385
Migrating agency criteria for agency 386
Migrating agency criteria for agency 384
Migrating agency criteria for agency 1
...
Migrating agency criteria for agency 383
Migrating agency criteria for users 12 belonging to agency 386
Migrating agency criteria for users 9,10,11 belonging to agency 113
Migrating agency criteria for users 1,2,3,4,5,17,18 belonging to agency 0
No agency criteria to migrate for users 14,15 belonging to agency 400
Migrating agency criteria for users 6,7,8,13 belonging to agency 384
No users to migrate for agency 1
...
Inserted 15 saved searches
Done migrating legacy agency criteria to saved searches
Done in 3.65s.
```

#### Before - Live Run
```
usdr_grants=# select count(*), array_agg(distinct created_by) from grants_saved_searches;
 count | array_agg 
-------+-----------
     4 | {4}
(1 row)
```

#### After - Live Run
```
usdr_grants=# select count(*), array_agg(distinct created_by) from grants_saved_searches;
 count |               array_agg               
-------+---------------------------------------
    19 | {1,2,3,4,5,6,7,8,9,10,11,12,13,17,18}
(1 row)
```

### Automated and Unit Tests
- [x] Added Unit tests

### Manual tests for Reviewer
- [x] Added steps to test feature/functionality manually

## Checklist
- [x] Provided ticket and description
- [x] Provided screenshots/demo
- [x] Provided testing information
- [x] Provided adequate test coverage for all new code
- [x] Added PR reviewers

### CODE REVIEW

File: packages/server/src/db/saved_search_migration.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+const { getKeywords } = require('.');
+const knex = require('./connection');
+
+async function main() {
+    await knex.transaction(async (trns) => {
+        console.log(trns);
+        // get agencies with keywords
```
Comment (as1729) [inline packages/server/src/db/saved_search_migration.js:7]: Need to get the eligibility codes as well

Comment (as1729) [reply packages/server/src/db/saved_search_migration.js]: Done 👍

**Review by TylerHendrickson — APPROVED** [review]
💯

**Comment by ClaireValdivia** [comment]
@as1729  can this be closed?

**Comment by as1729** [comment]
Yes @ClaireValdivia this issue can be closed.

**Pull request merged**